hypothesis-python/src/hypothesis/internal/cache.py

```diff
@@ -230,13 +232,12 @@ def __balance(self, i):
                 self.__swap(parent, i)
                 i = parent
             else:
-                # This branch is never taken on versions of Python where dicts
-                # preserve their insertion order (pypy or cpython >= 3.7)
-                break  # pragma: no cover
+                break
         while True:
             children = [j for j in (2 * i + 1, 2 * i + 2) if j < len(self.data)]
             if len(children) == 2:
-                children.sort(key=lambda j: self.data[j].score)
+                # try smallest child first
+                children.sort(key=lambda j: self.data[j].sort_key)
```

Comment:
This has been nagging on me today. We should only ever check the smallest child, right, not both? We only continue to the largest child we're in-order with (smaller than) the smallest child. And then we can't be out-of-order (larger than) the largest child. Yeah, a minor detail, but you know...